gk/fib.c

```diff
@@ -1332,11 +1362,17 @@ init_drop_fib_locked(struct ip_prefix *ip_prefix, uint8_t rt_proto,
 	return 0;
 }
 
+/*
+ * If a FIB entry already exists for @prefix, then
+ * @fib_id contains its index in the FIB table.
+ * Otherwise, @fib_id is <0.
+ */
 static int
 add_fib_entry_locked(struct ip_prefix *prefix,
 	struct ipaddr *gt_addrs, struct ipaddr *gw_addrs,
 	unsigned int num_addrs, enum gk_fib_action action,
-	uint8_t rt_proto, struct gk_config *gk_conf)
+	uint8_t rt_proto, struct gk_config *gk_conf,
+	int fib_id)
```

Comment:
`fib_id`  beign passed is ` uint32_t` (i.e. unsigned). We are better off using `int64_t`  here and in `init_grantor_fib_locked()`.